Insertion of architecture diagram and edits to the Architecture topic… (#373)
Mirantis/mke-docs: content/docs/concepts/architecture.md

## Components

> MKE 4k clusters are comprised of the following components:

Reviewer: ```suggestion
MKE 4k clusters consist of the following components:
```

- k0s Kubernetes distribution
- Container Network Interface (CNI)
- MKE 4k Control Plane
  - `mke-operator`
  - `kordent`
- System services
- User services

### k0s Kubernetes distribution

k0s is a lightweight and open-source Kubernetes distribution that acts as the
foundational runtime for MKE 4k clusters. For more information, refer to the

> [k0s documentation](https://docs.k0sproject.io/stable//).

Reviewer: ```suggestion
[k0s documentation](https://docs.k0sproject.io/stable).
```

### Container Network Interface (CNI)

> MKE 4k supports Container Network Interface (CNI) plugins, to provide

Reviewer: ```suggestion
MKE 4k supports Container Network Interface (CNI) plugins to provide
```

Author: Leaving as-is, for fluency.

networking function within a cluster that is secure, scalable, and
policy-driven.

> MKE 4k installs with the Calico CNI by default. Other CNI options, though, such

Reviewer: ```suggestion
MKE 4k installs with the Calico CNI by default, but other CNI options, such
```

Author: I prefer the original, simpler wording, which also serves to more strongly send the "Calico-is-default" message. Keeping as-is.

as Kube-router, are also available. Refer to [Container Network
Interface](../cni) for the network configuration details.

### MKE 4k Control Plane

> The MKE 4k Control Plane, which is comprised of `mke-operator` and `kordent`,

Reviewer: ```suggestion
The MKE 4k Control Plane, which comprises `mke-operator` and `kordent`,
```

is used to orchestrate, manage, and maintain the lifecycle of the MKE 4k
cluster and services.

- `mke-operator`: A Kubernetes-native operator that automates installation,
  upgrades, system configuration, and user services.
- `k0rdent`: An open-source component that performs lifecycle management for
  the services that `mke-operator` configures.

### System services

A key benefit of MKE 4k is that it includes various services that are typically

> standard to a cluster, thus ensuring that they integrate well and are easy to

Reviewer: ```suggestion
standard to a cluster, thus ensuring that they integrate well and are straightforward to
```

## Components

An MKE 4k cluster is comprised of the following components:

- k0s Kubernetes distribution
- Calico CNI
- MKE 4k Control Plane

> - System and user services

Reviewer 2: System and user services should be separated. We should also include k0rdent in this list. I'm also thinking `MKE 4k operator` makes more sense than control plane but I'm on the fence.

### k0s Kubernetes distribution

k0s is a lightweight and open-source Kubernetes distribution that acts as the
foundational runtime for MKE 4k clusters. For more information, refer to the
[k0s documentation](https://docs.k0sproject.io/stable//).

> ### Calico CNI

Reviewer 2: Calico is the default CNI. Others are available also. 

This makes saying `your cluster has calico` in a matter-of-facto way come off as incorrect to me.

MKE 4k provides secure, scalable, policy-driven networking through the Calico
container network interface (CNI).

> The Calico configuration at installation is as follows:

Reviewer 2: Do we have a calico page? lines 32-38 seem like details that belong on calico's page rather than the high level architecture page.

- IPv4 only, with a fixed Pod CIDR of `192.168.0.0/16`
- Datastore mode set to `kdd`
- `kube-proxy` mode set to `iptables`
- A `vxlan` backend that uses the default port of `4789` for traffic and the
  default virtual network ID of `4096`

Refer to [Container Network Interface](../cni) for the network configuration
details.

> ### MKE 4k Control Plane

Reviewer 2: Building on my comments above, I would almost say we should separate the `mke-operator` and `k0rdent` into separate sections as they are separate projects from one another.

The MKE 4k Control Plane, which is comprised of `mke-operator` and `kordent`,
is used to orchestrate, manage, and maintain the lifecycle of the MKE 4k
cluster and services.

- `mke-operator`: A Kubernetes-native operator that automates installation,
  upgrades, system configuration, and user services .
- `k0rdent`: An open-source component that performs lifecycle management for
  the services that `mke-operator` configures.

> ### System and User Services

Reviewer 2: I would separate these and hype up that MKE4k system services are services that are typically standard in a cluster (auth, ingress, monitoring, backups, etc) that we provide support for so that they are easy to configure and easy to use. These are really the bread and butter and benefits customers get by using mke4k so we should really emphasize how amazing they are. "much wow"

The user services section should talk about how customers can bring their workloads and worry about their stuff in an easy manner without worrying about the system service stuff. They just start building the things they really need without worrying about the default, system stuff that every cluster needs.

Also, we don't provide anything special for logging so we probably shouldn't list it

Refer to [Container Network Interface](../cni) for the network configuration
details.

### MKE 4k Control Plane

The MKE 4k Control Plane, which is comprised of `mke-operator` and `kordent`,
is used to orchestrate, manage, and maintain the lifecycle of the MKE 4k
cluster and services.

- `mke-operator`: A Kubernetes-native operator that automates installation,
  upgrades, system configuration, and user services .
- `k0rdent`: An open-source component that performs lifecycle management for
  the services that `mke-operator` configures.

### System and User Services

The system and user services are Kubernetes workloads that run within the
cluster, such as monitoring, logging, ingress, and user-defined applications.
Use the MKE 4 control plane to manage the deployment and lifecycle of these
workloads.

The system and user services are Kubernetes workloads that run within the MKE
4k cluster. On the system side, these services include monitoring, logging, and
ingress, among others. On the user side, the services are typically
user-defined applications.  Using the MKE 4 control plane, you can manage the
deployment and lifecycle of the various workloads.

> ## Architecture Diagram

Reviewer 2: The svg should be in the `static/images` folder. Then you can reference it with `mke-docs/mke4-architecture.drawio.svg`

> - `k0rdent`: An open-source component that performs lifecycle management for

Reviewer 2: Since we split this, we should put something for the mke-operator too

Author: Hmm. Not getting it. The line above the one highlighted is:

- `mke-operator`: A Kubernetes-native operator that automates installation, upgrades, system configuration, and user services.

Reviewer 2: I think github was covering it up. I didn't see the line above before

  the services that `mke-operator` configures.

### System services

> A key benefit of MKE 4k is that it supports various system services that are

Reviewer 2: Just a little rewording

```
A key benefit of MKE 4k is that it includes various services that are
typically standard to a cluster, thus ensuring they are easy to use, configure, and integrate with one another.
```

Author: For greater fluency, let's go with:

A key benefit of MKE 4k is that it includes various services that are typically standard to a cluster, thus ensuring that they integrate well and are easy to use and configure.

Reviewer 2: That sounds great to me

## Architecture Diagram

> ![mke4-architecture.drawio.svg](mke4-architecture.drawio.svg)

Reviewer 2: For a modal, replace this line with the following

```

<img src="/mke-docs/mke4-architecture.drawio.svg" id="myBtn"></img>

<div id="myModal" class="modal">
  <div class="modal-content">
    <span class="close">&times;</span>
    <img src="/mke-docs/mke4-architecture.drawio.svg">
  </div>
</div>

<script>
var modal = document.getElementById("myModal");
var btn = document.getElementById("myBtn");
var span = document.getElementsByClassName("close")[0];

// When the user clicks the button, open the modal
btn.onclick = function() {
  modal.style.display = "block";
}

// When the user clicks on <span> (x), close the modal
span.onclick = function() {
  modal.style.display = "none";
}

// When the user clicks anywhere outside of the modal, close it
window.onclick = function(event) {
  if (event.target == modal) {
    modal.style.display = "none";
  }
}
</script>

<style>
.modal {
  display: none; /* Hidden by default */
  position: fixed; /* Stay in place */
  z-index: 1; /* Sit on top */
  padding-top: 100px; /* Location of the box */
  left: 0;
  top: 0;
  width: 100%; /* Full width */
  height: 100%; /* Full height */
  overflow: auto; /* Enable scroll if needed */
  background-color: rgb(0,0,0); /* Fallback color */
  background-color: rgba(0,0,0,0.4); /* Black w/ opacity */
}

/* Modal Content */
.modal-content {
  background-color: #fefefe;
  margin: auto;
  padding: 20px;
  border: 1px solid #888;
  width: 80%;
}

/* The Close Button */
.close {
  color: #aaaaaa;
  float: right;
  font-size: 28px;
  font-weight: bold;
}

.close:hover,
.close:focus {
  color: #000;
  text-decoration: none;
  cursor: pointer;
}
</style>
```

I'll let you do any fine tuning

Author: Inserted `/images/` to img src, as worked out with @nwneisen via Slack.
